virtual_rainforest/core/utils.py

```diff
@@ -59,62 +61,91 @@ def check_outfile(merge_file_path: Path) -> None:
     return None
 
 
-def set_layer_roles(canopy_layers: int, soil_layers: int) -> list[str]:
+def set_layer_roles(canopy_layers: int, soil_layers: list[float]) -> list[str]:
     """Create a list of layer roles.
 
-    This function creates a list of layer roles for the vertical dimension of the
-    Virtual Rainforest. The layer above the canopy is defined as 0 (canopy height + 2m)
-    and the index increases towards the bottom of the soil column. The canopy includes a
-    maximum number of canopy layers (defined in config) which are filled from the top
-    with canopy node heights from the plant module (the rest is set to NaN). Below the
-    canopy, we currently set one subcanopy layer (around 1.5m above ground) and one
-    surface layer (0.1 m above ground). Below ground, we include a maximum number of
-    soil layers (defined in config); the deepest layer is currently set to 1 m as the
-    temperature there is fairly constant and equals the mean annual temperature.
+    This function creates a list of strings describing the layer roles for the vertical
+    dimension of the Virtual Rainforest. The vertical dimension consists of the
+    following layers and roles.
+
```

Comment:
maybe add here a comment that above ground values are positive and below ground negative